packages/ra-core/src/controller/edit/useEditController.spec.tsx

```diff
@@ -224,17 +224,21 @@ describe('useEditController', () => {
         let saveCallback;
         render(
             <CoreAdminContext dataProvider={dataProvider}>
-                <SaveContextProvider value={saveContextValue}>
-                    <EditController
-                        {...defaultProps}
-                        mutationMode="pessimistic"
-                    >
-                        {({ save, record }) => {
-                            saveCallback = save;
-                            return <>{JSON.stringify(record)}</>;
-                        }}
-                    </EditController>
-                </SaveContextProvider>
+                <EditController {...defaultProps} mutationMode="pessimistic">
+                    {({ save, saving, record }) => {
+                        saveCallback = save;
+                        return (
+                            <SaveContextProvider
```

Comment:
you don't need a SaveContext for this test. Same for some of the following tests